### PR TITLE
Sparse vector grpc definition

### PR DESCRIPTION
**Pull request by IvanPleshkov**
Add optional field to `Vector` grpc structure. Added conversion and fixed recommendation example (was required for proper conventions). Added fields into `SearchPoints` grpc definition where `vector` field is just a `Vec<f32>`.

TODO:
- validation of `SearchPoints` and `SearchPointGroups`

### All Submissions:

* [ ] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [ ] Have you followed the guidelines in our Contributing document?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [ ] Does your submission pass tests?
2. [ ] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
3. [ ] Have you checked your code using `cargo clippy --all --all-features` command?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: lib/api/src/grpc/proto/points.proto [hunk 1]
```diff
@@ -41,8 +41,13 @@ message PointId {
   }
 }
 
+message ValuesIndices {
+  repeated uint32 data = 1;
+}
+
 message Vector {
   repeated float data = 1;
+  optional ValuesIndices indices = 2;
```
Comment (agourlay) [inline lib/api/src/grpc/proto/points.proto:50]: will this encoding allow empty sparse vectors?
empty values & empty indices/

**Comment by agourlay** [comment]
Regarding the todo

> validation of SearchPoints and SearchPointGroups

Can we maybe rely on the `SparseVector` validation to detect malformed request?

File: lib/api/src/grpc/proto/points.proto (hunk 1 above)
Comment (IvanPleshkov) [reply lib/api/src/grpc/proto/points.proto:50]: It seems empty sparse vector is avaliable. For empty sparse vector `ValuesIndices` must be presented and `ValuesIndices.data` must be empty

**Comment by IvanPleshkov** [comment]
> Regarding the todo
> 
> > validation of SearchPoints and SearchPointGroups
> 
> Can we maybe rely on the `SparseVector` validation to detect malformed request?

I didn't get your point. Currently I reuse validation check for both rest and grpc apis

File: lib/api/src/grpc/proto/points.proto [hunk 2]
```diff
@@ -258,6 +263,7 @@ message SearchPoints {
   optional ReadConsistency read_consistency = 12; // Options for specifying read consistency guarantees
   optional uint64 timeout = 13; // If set, overrides global timeout setting for this request. Unit is seconds.
   optional ShardKeySelector shard_key_selector = 14; // Specify in which shards to look for the points, if not specified - look in all shards
+  optional ValuesIndices indices = 15;
```
Comment (generall) [inline lib/api/src/grpc/proto/points.proto:266]: sparce_indices

File: lib/api/src/grpc/proto/points.proto [hunk 3]
```diff
@@ -290,6 +296,7 @@ message SearchPointGroups {
   optional WithLookup with_lookup = 13; // Options for specifying how to use the group id to lookup points in another collection
   optional uint64 timeout = 14; // If set, overrides global timeout setting for this request. Unit is seconds.
   optional ShardKeySelector shard_key_selector = 15; // Specify in which shards to look for the points, if not specified - look in all shards
+  optional ValuesIndices indices = 16;
```
Comment (generall) [inline lib/api/src/grpc/proto/points.proto:299]: sparce_indices

File: lib/api/src/grpc/proto/points.proto [hunk 4]
```diff
@@ -41,8 +41,13 @@ message PointId {
   }
 }
 
+message ValuesIndices {
```
Comment (generall) [inline lib/api/src/grpc/proto/points.proto:44]: SparseIndices

Comment (IvanPleshkov) [reply lib/api/src/grpc/proto/points.proto]: fixed

File: lib/api/src/grpc/proto/points.proto (hunk 3 above)
Comment (IvanPleshkov) [reply lib/api/src/grpc/proto/points.proto]: fixed

File: lib/api/src/grpc/proto/points.proto (hunk 2 above)
Comment (IvanPleshkov) [reply lib/api/src/grpc/proto/points.proto]: fixed

**Comment by agourlay** [comment]
@IvanPleshkov do we need a `TODO(sparse)` for the missing validation?

**Comment by IvanPleshkov** [comment]
> @IvanPleshkov do we need a `TODO(sparse)` for the missing validation?

done

**Review by agourlay — APPROVED** [review]

**Review by generall — APPROVED** [review]

**Pull request merged**